litestar/controller.py

```diff
@@ -209,7 +209,9 @@ def get_route_handlers(self) -> list[BaseRouteHandler]:
         self_handlers.sort(key=attrgetter("handler_id"))
         for self_handler in self_handlers:
             route_handler = deepcopy(self_handler)
-            route_handler.fn.value = partial(route_handler.fn.value, self)
+            # at the point we get a reference to the handler function, it's unbound, so
+            # we replace it with a regular bound method here
+            route_handler._fn = types.MethodType(route_handler._fn, self)
```

Comment:
This has probably the same outcome in most cases but is technically more correct.

We should improve the way controllers are registered in general so that this + the `deepcopy` isn't necessary. Afaik there's nothing fundamentally preventing this.